flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/AdaptedRestartPipelinedRegionStrategyNG.java

```diff
@@ -108,8 +106,7 @@ protected void restartTasks(final Set<ExecutionVertexID> verticesToRestart) {
 
 		FutureUtils.assertNoException(
 			cancelTasks(verticesToRestart)
-				.thenRunAsync(resetAndRescheduleTasks(globalModVersion, vertexVersions), executionGraph.getJobMasterMainThreadExecutor())
-				.handle(failGlobalOnError()));
```

Comment:
I think we should not change this logic since `cancelTasks` can return an exceptionally completed future which would now cause the process to terminate.

Reply:
Agreed, I am refactoring this PR by changing the result of `RestartStrategy#restart` to a `CompletableFuture<?>`